### PR TITLE
Improve RegressionExpeirment using AutoMLExperiment

### PR DESCRIPTION
**Pull request by LittleLittleCloud**
We are excited to review your PR.

So we can do the best job, please check:

- [x] There's a descriptive title that will make sense to other developers some time from now. 
- [x] There's associated issues. All PR's should have issue(s) associated - unless a trivial self-evident change such as fixing a typo. You can use the format `Fixes #nnnn` in your description to cause GitHub to automatically close the issue(s) when your PR is merged.
- [x] Your change description explains what the change does, why you chose your approach, and anything else that reviewers should know.
- [x] You have included any necessary tests in the same PR.

- https://github.com/dotnet/machinelearning/issues/6145

### CODE REVIEW

File: test/Microsoft.ML.AutoML.Tests/AutoFitTests.cs [hunk 1]
```diff
@@ -277,55 +368,6 @@ public void AutoFitMultiClassification_Image()
             Assert.Equal(TextDataViewType.Instance, scoredData.Schema[DefaultColumnNames.PredictedLabel].Type);
         }
 
-        [Theory]
-        [InlineData("en-US")]
-        [InlineData("ar-SA")]
-        [InlineData("pl-PL")]
-        public void AutoFitRegressionTest(string culture)
```
Comment (LittleLittleCloud) [inline test/Microsoft.ML.AutoML.Tests/AutoFitTests.cs:284]: We no longer using SMac Sweeper in regression, so we can just remove this test.

File: src/Microsoft.ML.AutoML/API/MulticlassClassificationExperiment.cs [hunk 2]
```diff
@@ -159,7 +159,7 @@ public override ExperimentResult<MulticlassClassificationMetrics> Execute(IDataV
             // split cross validation result according to sample key as well.
             if (rowCount < crossValRowCountThreshold)
             {
-                const int numCrossValFolds = 10;
+                int numCrossValFolds = 10;
```
Comment (tarekgh) [inline src/Microsoft.ML.AutoML/API/MulticlassClassificationExperiment.cs:162]: why you have removed the const?

File: src/Microsoft.ML.AutoML/API/RegressionExperiment.cs [hunk 3]
```diff
@@ -132,6 +131,187 @@ internal RegressionExperiment(MLContext context, RegressionExperimentSettings se
...
+        {
+            var label = columnInformation.LabelColumnName;
+            _experiment.SetRegressionMetric(Settings.OptimizingMetric, label);
+
+            // Cross val threshold for # of dataset rows --
+            // If dataset has < threshold # of rows, use cross val.
+            // Else, run experiment using train-validate split.
+            const int crossValRowCountThreshold = 15000;
+            var rowCount = DatasetDimensionsUtil.CountRows(trainData, crossValRowCountThreshold);
+            // TODO
```
Comment (tarekgh) [inline src/Microsoft.ML.AutoML/API/RegressionExperiment.cs:154]: >TODO [](http://example.com/codeflow?start=15&length=4)

this will be done in another PR?

**Review by tarekgh — APPROVED** [review]
module my minor questions, LGTM.

File: src/Microsoft.ML.AutoML/API/RegressionExperiment.cs (hunk 3 above)
Comment (LittleLittleCloud) [reply src/Microsoft.ML.AutoML/API/RegressionExperiment.cs:154]: Yep

File: src/Microsoft.ML.AutoML/API/MulticlassClassificationExperiment.cs (hunk 2 above)
Comment (LittleLittleCloud) [reply src/Microsoft.ML.AutoML/API/MulticlassClassificationExperiment.cs:162]: There's no specific reason of doing that. I just feel like there's no need to mark this `numCrossValFolds` as const.

Comment (tarekgh) [reply src/Microsoft.ML.AutoML/API/MulticlassClassificationExperiment.cs:162]: why you have a variable in the first place then? I mean why you don't just call the method and passing `10`?

Comment (LittleLittleCloud) [reply src/Microsoft.ML.AutoML/API/MulticlassClassificationExperiment.cs:162]: I didn't notice that,, fixed and Resolved

File: src/Microsoft.ML.AutoML/API/MulticlassClassificationExperiment.cs [hunk 4]
```diff
@@ -159,8 +159,7 @@ public override ExperimentResult<MulticlassClassificationMetrics> Execute(IDataV
             // split cross validation result according to sample key as well.
             if (rowCount < crossValRowCountThreshold)
             {
-                const int numCrossValFolds = 10;
-                _experiment.SetDataset(trainData, numCrossValFolds);
+                _experiment.SetDataset(trainData, 10);
```
Comment (tarekgh) [inline src/Microsoft.ML.AutoML/API/MulticlassClassificationExperiment.cs:162]: nit: you may add the parameter name before `10` for code readability :-)
